src/module/item/consumable/document.ts

```diff
@@ -205,33 +207,30 @@ class ConsumablePF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extend
     }
 
     async castEmbeddedSpell(trickMagicItemData?: TrickMagicItemEntry): Promise<void> {
-        const { actor } = this;
+        const actor = this.actor;
         const spell = this.embeddedSpell;
         if (!actor || !spell) return;
 
         // Find the best spellcasting entry to cast this consumable
-        const entry = (() => {
+        const entry = ((): SpellcastingEntry<ActorPF2e> | null => {
             if (trickMagicItemData) return trickMagicItemData;
-            return actor.spellcasting
-                .filter(
-                    (e): e is SpellcastingEntryPF2e<ActorPF2e> =>
-                        e instanceof SpellcastingEntryPF2e && e.canCast(spell, { origin: this }),
-                )
-                .reduce((previous, current) => {
-                    const previousDC = previous.statistic.dc.value;
-                    const currentDC = current.statistic.dc.value;
-                    return currentDC > previousDC ? current : previous;
-                });
+            return (
+                actor.spellcasting
+                    .filter(
+                        (e): e is SpellcastingEntry<ActorPF2e> & { statistic: Statistic } =>
+                            !!e.statistic && e.canCast(spell, { origin: this }),
+                    )
+                    .reduce((previous, current) => {
+                        if (!previous && current?.statistic) return current;
+                        if (!current?.statistic && previous?.statistic) return previous;
```

Comment:
Wouldn't the below defaults to negative infinity render these if statements unnecessary?